packages/http-client-java/generator/http-client-generator-test/src/test/java/azure/resourcemanager/operationtemplates/OperationTests.java

```diff
@@ -86,6 +94,41 @@ public void testLro() {
         manager.lroes().deleteById(order.id());
     }
 
+    @Test
+    public void testOptionalBody() {
+        String rgName = "test-rg";
+        String resourceName = "widget1";
+        Widget widget = manager.optionalBodies().getByResourceGroup(rgName, resourceName);
+        Assertions.assertEquals("A test widget", widget.properties().description());
+        widget = manager.optionalBodies().patch(rgName, resourceName);
+        Assertions.assertEquals("A test widget", widget.properties().description());
+
+        ChangeAllowanceResult result = manager.optionalBodies().providerPost();
+        Assertions.assertEquals(50, result.totalAllowed());
+        Assertions.assertEquals("Changed to default allowance", result.status());
+        result = manager.optionalBodies()
+            .providerPostWithResponse(new ChangeAllowanceRequest().withReason("Increased demand").withTotalAllowed(100),
```

Comment:
Why it calls `providerPostWithResponse` instead of `providerPost`?

Similar for other ops.